fiftyone/core/collections.py

```diff
@@ -440,6 +442,76 @@ def _get_default_frame_fields(
             include_private=include_private, use_db_fields=use_db_fields
         )
 
+    def get_field(self, path):
```

Comment:
`SampleCollection.get_field()` is the new method for accessing any `Field` instance on a schema via its path.

```py
import fiftyone.zoo as foz

quickstart = foz.load_zoo_dataset("quickstart")
print(quickstart.get_field("ground_truth.detections.confidence"))

video_quickstart = foz.load_zoo_dataset("quickstart-video")
print("frames.detections.detections", video_quickstart.get_field("frames.detections.detections")
```

One point I did not address is dynamic `document_type` updates. Perhaps this should be addressed at some point. The obvious example relates to metadata:
```py
quickstart.compute_metadata(num_workers=1, skip_failures=False) # multiprocessing isn't working...

metadata_field = quickstart.get_field("metadata")
print(metadata_field) # fiftyone.core.Metadata, not fiftyone.core.metadata.ImageMetadata
```